src/librustc_trait_selection/traits/error_reporting/mod.rs

```diff
@@ -283,6 +283,8 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
                             .unwrap_or(false);
                         let is_from = format!("{}", trait_ref.print_only_trait_path())
                             .starts_with("std::convert::From<");
+                        let is_unsize =
+                            { Some(trait_ref.def_id()) == self.tcx.lang_items().unsize_trait() };
```

Comment:
Are the braces needed? Have you ran `x.py fmt`?

Reply:
Not strictly necessary, but I ran `./x.py fmt`

Reply:
I can remove if you think that's better?

Reply:
Na, if it's fine with fmt it's fine with me.